state-sync/storage-service/server/src/tests/utils.rs

```diff
@@ -429,3 +432,19 @@ pub async fn wait_for_optimistic_fetch_service_to_refresh(
     // Elapse enough time to force the optimistic fetch thread to work
     advance_storage_refresh_time(mock_time).await;
 }
+
+/// Waits for the specified number of optimistic fetches to be active
+pub async fn wait_for_active_optimistic_fetches(
+    active_optimistic_fetches: Arc<Mutex<HashMap<PeerNetworkId, OptimisticFetchRequest>>>,
+    expected_num_active_fetches: usize,
+) {
+    loop {
+        let num_active_fetches = active_optimistic_fetches.lock().len();
+        if num_active_fetches == expected_num_active_fetches {
```

Comment:
Is it always an exact match? If not, >=

Reply:
If yes, handle the case for >.

Reply:
Yeah, it's always an exact match 😄 Anything else would be a bug in the code.